app/Main.hs

```diff
@@ -494,29 +495,39 @@ data BowerOutput = BowerOutput
 addFromBower :: String -> IO ()
 addFromBower name = do
   let bowerProc = inproc "bower" [ "info", T.pack name, "--json", "-l=error" ] empty
-  result <- fold <$> shellToIOText bowerProc
-  if T.null result
-    then exitWithErr "Error: Does the package exist on Bower?"
-    else do
-      let result' = do
-            bowerOutput <- Aeson.eitherDecodeStrict $ encodeUtf8 result
-            let bowerInfo = latest bowerOutput
-            pkgName <- mkPackageName' $ bower_name bowerInfo
-            packageNames <- traverse mkPackageName' $ Map.keys (bower_dependencies bowerInfo)
-            pure $
-              ( pkgName
-              , PackageInfo
-                (url $ bower_repository bowerInfo)
-                ("v" <> bower_version bowerInfo)
-                packageNames
-              )
-      case result' of
-        Right (pkgName, info) -> do
-          pkg <- readPackageFile
-          db <- readPackageSet pkg
-          writePackageSet pkg $ Map.insert pkgName info db
-          echoT $ "Successfully wrote " <> runPackageName pkgName <> " to package set."
-        Left errors -> echoT $ "Errors processing Bower Info: " <> (T.pack errors)
+  (eitherResult :: Either IOError Text) <- tryIOError $ fold <$> shellToIOText bowerProc
+  case eitherResult of
+    Left err' ->
+      if List.isInfixOf "does not exist" $ ioeGetErrorString err'
```

Comment:
If you wanted to do this check though it probably needs to not be hardcoded like this though. Maybe running `where bower` would be better for this